src/main/scala/model/Kinds.scala

```diff
@@ -0,0 +1,113 @@
+
+import scala.language.implicitConversions
+import scala.util.Random
+
+/**
+ * An Enumeration for all the Alleles present in the World.
+ */
+object Alleles extends Enumeration{
+  type AlleleKind = Value
+
+  /**
+   * The information each AlleleKind must have
+   * @param dominant specifies if it's dominant or not, it could be empty it not chosen yet
+   * @param locked   show if the dominance has already been chosen (becoming unmodifiable) or not
+   */
+  protected case class AllelesVal(private var dominant: Option[Boolean] = Option.empty,
+                                  var locked: Boolean = false) extends super.Val {
+    def setDominance(cond: Boolean): Unit = {
+      if (locked) throw new MultipleDominanceAssignmentException else {
+        dominant = Option(cond)
+        locked = true
+      }
+    }
+    def isDominant: Option[Boolean] = dominant
+  }
+  implicit def valueToAllelesVal(x: Value): AllelesVal = x.asInstanceOf[AllelesVal]
+
+  val WHITE_FUR: AllelesVal = AllelesVal()
+  val BROWN_FUR: AllelesVal = AllelesVal()
+  val LONG_FUR: AllelesVal = AllelesVal()
+  val SHORT_FUR: AllelesVal = AllelesVal()
+  val LONG_TEETH: AllelesVal = AllelesVal()
+  val SHORT_TEETH: AllelesVal = AllelesVal()
+  val HIGH_EARS: AllelesVal = AllelesVal()
+  val LOW_EARS: AllelesVal = AllelesVal()
+  val HIGH_JUMP: AllelesVal = AllelesVal()
+  val LOW_JUMP: AllelesVal = AllelesVal()
+}
+
+/**
+ * An Enumeration for all the Genes present in the World, which every Bunny must have.
+ */
+object Genes extends Enumeration {
+  type GeneKind = Value
+  import Alleles.AlleleKind
+
+  /**
+   * The information each GeneKind must have.
+   * @param base    the base Allele of the gene
+   * @param mutated the mutated Allele of the gene
+   * @param letter  the letter which corresponds to this gene
+   */
+  protected case class GenesVal(base: AlleleKind,
+                                mutated: AlleleKind,
+                                letter: String) extends super.Val
+  import scala.language.implicitConversions
+  implicit def valueToGenesVal(x: Value): GenesVal = x.asInstanceOf[GenesVal]
+
+  val FUR_COLOR: GenesVal =  GenesVal(base = Alleles.WHITE_FUR,
+                                mutated = Alleles.BROWN_FUR,
+                                letter = "f")
+  val FUR_LENGTH: GenesVal = GenesVal(base = Alleles.SHORT_FUR,
+                                mutated = Alleles.LONG_FUR,
+                                letter = "l")
+  val TEETH: GenesVal =      GenesVal(base = Alleles.SHORT_TEETH,
+                                mutated = Alleles.LONG_TEETH,
+                                letter = "t")
+  val EARS: GenesVal =       GenesVal(base = Alleles.HIGH_EARS,
+                                mutated = Alleles.LOW_EARS,
+                                letter = "e")
+  val JUMP: GenesVal =       GenesVal(base = Alleles.LOW_JUMP,
+                                mutated = Alleles.HIGH_JUMP,
+                                letter = "j")
```

Comment:
Stesso discorso anche per queste costanti: io in questo caso le lascerei qui in quanto parte integrante dell'object _Genes_